sys/shell/commands/sc_openwsn.c

```diff
@@ -0,0 +1,566 @@
+        if (id == components[i].id) {
+            return components[i].name;
+        }
+    }
+    return NULL;
+}
+
+int _openwsn_ifconfig(char *arg)
+{
+    (void)arg;
+
+    open_addr_t *hwaddr;
+    open_addr_t temp_my128bID;
+
+    memcpy(&temp_my128bID.addr_128b[0], &idmanager_vars.myPrefix.prefix, 8);
+    memcpy(&temp_my128bID.addr_128b[8], &idmanager_vars.my64bID.addr_64b, 8);
+
+    printf("Iface  %d  ", openwsn_get_pid());
+    hwaddr = idmanager_getMyID(ADDR_16B);
+    printf("\tHWaddr: %s  ", netif_addr_to_str(hwaddr->addr_16b, 2, _addr_str));
+    hwaddr = idmanager_getMyID(ADDR_PANID);
+    printf("NID: %s\n\n", netif_addr_to_str(hwaddr->panid, 2, _addr_str));
+
+    hwaddr = idmanager_getMyID(ADDR_64B);
+    printf("\t\tLong HWaddr: %s\n",
+           netif_addr_to_str(hwaddr->addr_64b, 8, _addr_str));
+
+    if(IS_USED(ODULE_OPENWSN_IPV6)) {
+        ipv6_addr_to_str(_addr_str, (ipv6_addr_t *)temp_my128bID.addr_128b,
+                        sizeof(_addr_str));
+        printf("\t\tinet6 addr: %s\n\n", _addr_str);
+    }
+
+    printf("\t\tIEEE802154E sync: %i\n", ieee154e_isSynch());
+    if (IS_USED(MODULE_OPENWSN_CJOIN)) {
+        printf("\t\t6TiSCH joined: %i\n", cjoin_getIsJoined());
+    }
+    puts("");
+
+    if (IS_USED(MODULE_OPENWSN_IPV6)) {
+        extern icmpv6rpl_vars_t icmpv6rpl_vars;
+        if (idmanager_vars.isDAGroot) {
+            puts("\t\tNode is DAG root");
+        }
+        else {
+            if (icmpv6rpl_vars.haveParent) {
+                printf("\t\tRPL rank: %i\n", icmpv6rpl_vars.myDAGrank);
+                printf("\t\tRPL parent: %s\n", \
+                    netif_addr_to_str(neighbors_vars.neighbors[icmpv6rpl_vars.
+                                                                ParentIndex].
+                                        addr_64b.addr_64b, 8, _addr_str));
+                printf("\t\tRPL children:\n");
+                for (uint8_t i = 0; i < MAXNUMNEIGHBORS; i++) {
+                    if (neighbors_isNeighborWithHigherDAGrank(i)) {
+                        printf("\t\t\t%s\n", \
+                            netif_addr_to_str(neighbors_vars.neighbors[i].
+                                                addr_64b.addr_64b, 8, _addr_str));
+                    }
+                }
+                ipv6_addr_to_str(_addr_str,
+                                (ipv6_addr_t *)icmpv6rpl_vars.dao.DODAGID,
+                                sizeof(_addr_str));
+                printf("\t\tRPL DODAG ID: %16s\n", _addr_str);
+            }
+            else {
+                puts("\t\tNO RPL parent");
+            }
+        }
+    }
+
+    return 0;
+}
+
+static int _neighbors_cmd(char *arg)
+{
+    (void)arg;
+
+    for (int i = 0; i < MAXNUMNEIGHBORS; i++) {
+        netif_addr_to_str(neighbors_vars.neighbors[i].addr_64b.addr_64b, 8,
+                          _addr_str);
+        if (memcmp(_addr_str, "00:00:00:00:00:00:00:00", 8) != 0) {
```

Comment:
Here the 8 is the size of the string to compare but is not correct, wouldn't it be more robust to use `IPV6_ADDR_MAX_STR_LEN` instead ?
```suggestion
        if (memcmp(_addr_str, "00:00:00:00:00:00:00:00", IPV6_ADDR_MAX_STR_LEN) != 0) {
```

Reply:
If this makes sense, this is my last comment code-wise. Please squash after this is addressed (or answered).

Reply:
In this string looks like a MAC address. So why using IPV6_ADDR_MAX_STR_LEN as the size of `_addr_str` ?

Reply:
I was reusing the buffer for ipv6 addresses as well, I'll remove it.

Reply:
Would having separate buffers for both be best?

Reply:
I'll replace all occurences

Reply:
I know remember I used IPV6_ADDR_MAX_STR_LEN because there is not equivalent for mac addresses. But its true this might be confusing, the thing is that in cases where I print both the `MAC` and `IPV` address I would still rather re-use the same buffer, and `IPV6_ADDR_MAX_STR_LEN` will cover both.